File: app/src/main/java/pl/mosenko/sunnypodlaskie/mvp/weatherdatalist/WeatherDataListActivity.java
package pl.mosenko.sunnypodlaskie.mvp.weatherdatalist;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;

import pl.mosenko.sunnypodlaskie.R;
import pl.mosenko.sunnypodlaskie.mvp.setting.SettingsActivity;
import pl.mosenko.sunnypodlaskie.mvp.weatherdatadetail.WeatherDataDetailFragment;
import pl.mosenko.sunnypodlaskie.mvp.weatherdatadetail.WeatherDataDetailsActivity;

public class WeatherDataListActivity extends AppCompatActivity implements WeatherDataListFragment.Callback {

    private boolean twoPane;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initializeActivity();
        setupTwoPane();
    }

    private void setupTwoPane() {
        if (findViewById(R.id.WeatherDataListFragment_FrameLayout_container) != null) {
            twoPane = true;
        }
    }

    private void initializeActivity() {
        setContentView(R.layout.activity_weather_data_list);
        getSupportActionBar().setElevation(0f);
    }

    @Override
    protected void onStop() {
        super.onStop();
    }

    @Override
    public void onItemSelected(long weatherDataId) {
        if (twoPane) {
            replaceWeatherDataDetailsFragment(weatherDataId);
        } else {
            startWeatherDataDetailsActivity(weatherDataId);
        }
    }

    private void startWeatherDataDetailsActivity(long weatherDataId) {
        Intent intent = new Intent(this, WeatherDataDetailsActivity.class);
        intent.putExtra(WeatherDataDetailFragment.ARG_WEATHER_DATA_ID, weatherDataId);
        startActivity(intent);
    }

    private void replaceWeatherDataDetailsFragment(long weatherDataId) {
        Bundle arguments = new Bundle();
        arguments.putLong(WeatherDataDetailFragment.ARG_WEATHER_DATA_ID, weatherDataId);
        WeatherDataDetailFragment fragment = new WeatherDataDetailFragment();
        fragment.setArguments(arguments);
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.WeatherDataListFragment_FrameLayout_container, fragment)
                .commit();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater menuInflater = getMenuInflater();
        menuInflater.inflate(R.menu.main_menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        if (id == R.id.WeatherDataListActivity_action_settings) {
            startActivity(new Intent(this, SettingsActivity.class));
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/pl/mosenko/sunnypodlaskie/mvp/weatherdatalist/WeatherDataListActivity.kt
package pl.mosenko.sunnypodlaskie.mvp.weatherdatalist

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.*
import pl.mosenko.sunnypodlaskie.R
import pl.mosenko.sunnypodlaskie.mvp.setting.SettingsActivity
import pl.mosenko.sunnypodlaskie.mvp.weatherdatadetail.WeatherDataDetailFragment
import pl.mosenko.sunnypodlaskie.mvp.weatherdatadetail.WeatherDataDetailsActivity

class WeatherDataListActivity : AppCompatActivity(), WeatherDataListFragment.Callback {
    private var twoPane = false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initializeActivity()
        setupTwoPane()
    }

    private fun setupTwoPane() {
        if (findViewById<View?>(R.id.fl_container) != null) {
            twoPane = true
        }
    }

    private fun initializeActivity() {
        setContentView(R.layout.activity_weather_data_list)
        supportActionBar?.elevation = 0f
    }

    override fun onItemSelected(weatherDataId: Long) {
        if (twoPane) {
            replaceWeatherDataDetailsFragment(weatherDataId)
        } else {
            startWeatherDataDetailsActivity(weatherDataId)
        }
    }

    private fun startWeatherDataDetailsActivity(weatherDataId: Long) {
        val intent = Intent(this, WeatherDataDetailsActivity::class.java)
        intent.putExtra(WeatherDataDetailFragment.ARG_WEATHER_DATA_ID, weatherDataId)
        startActivity(intent)
    }

    private fun replaceWeatherDataDetailsFragment(weatherDataId: Long) {
        val arguments = Bundle()
        arguments.putLong(WeatherDataDetailFragment.ARG_WEATHER_DATA_ID, weatherDataId)
        val fragment = WeatherDataDetailFragment()
        fragment.arguments = arguments
        supportFragmentManager.beginTransaction()
                .replace(R.id.fl_container, fragment)
                .commit()
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        val menuInflater = menuInflater
        menuInflater.inflate(R.menu.main_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId
        if (id == R.id.action_settings) {
            startActivity(Intent(this, SettingsActivity::class.java))
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}
